### ISSUE TITLE
Couldn't connect to socket

### ISSUE THREAD

**Issue by rmasri79** [haugene/docker-transmission-openvpn#849]
Hello,
Thank you very much for the amazing image and great work.   I'm new to containers so my troubleshooting skills are still beginner level.  I'm running your image on Ubuntu 18 with Torguard as the VPN provider, but far I haven't been able to get transmission to download.  My transmission logs are filling up with these errors: 
2019-07-06 23:57:53.859] Couldn't connect socket 32 to 2600:380:a329:7168:bd9e:df63:162b:ed60, port 50814 (errno 99 - Cannot assign requested address) (net.c:290)
[2019-07-06 23:58:02.859] Couldn't connect socket 24 to 2001:8003:2b76:8900:48fb:6915:3721:1912, port 6500 (errno 99 - Cannot assign requested address) (net.c:290)
[2019-07-06 23:58:02.859] Couldn't connect socket 24 to 2001:8003:2b76:8900:48fb:6915:3721:1912, port 6500 (errno 99 - Cannot assign requested address) (net.c:290)
Not sure if this error is due to the fact that Torguard is now adding support for IPv6 or if its something else.  I disabled IPv6 on the host but not sure where else to disable it. Looking at the container logs I see the following:
Sun Jul  7 00:00:37 2019 Could not determine IPv4/IPv6 protocol

I'm not able to find anything about this online so I'm hoping someone can point me in the right direction.

Thanks

**Comment by haugene** [comment]
Please post your `docker run` command or docker-compose file so that I can see how you're running this. Otherwise, for the IPv6 part you could have a look at #847

**Comment by rmasri79** [comment]
Thank you very much haugene for replying so fast my docker compose file is as follows:
version: "3.6"
services:
# Transmission with VPN – Bittorrent Downloader
  transmission-vpn:
    container_name: transmission-vpn
    image: haugene/transmission-openvpn
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun
    restart: always
    ports:
    - "9091:9091"
    dns:
      - 10.9.0.1
      - 10.8.0.1
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ${USERDIR}/docker/transmission-vpn:/data
      - ${USERDIR}/docker/shared:/shared
      - ${USERDIR}/Downloads:/data/watch
      - ${USERDIR}/Downloads/completed:/data/completed
      - ${USERDIR}/Downloads/incomplete:/data/incomplete
    environment:
      - OPENVPN_PROVIDER=TORGUARD
      - OPENVPN_USERNAME=xxxx
      - OPENVPN_PASSWORD=yyyy
      - OPENVPN_CONFIG=Toronto 
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - LOCAL_NETWORK=192.168.1.0/24
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
      - TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true
      - TRANSMISSION_RPC_HOST_WHITELIST="127.0.0.1,192.168.*.*"
      - TRANSMISSION_RPC_PASSWORD=xxx
      - TRANSMISSION_RPC_USERNAME=yyy
      - TRANSMISSION_UMASK=002
      - TRANSMISSION_RATIO_LIMIT=1.00
      - TRANSMISSION_RATIO_LIMIT_ENABLED=true
      - TRANSMISSION_ALT_SPEED_DOWN=0
      - TRANSMISSION_ALT_SPEED_ENABLED=true
      - TRANSMISSION_ALT_SPEED_TIME_BEGIN=420
      - TRANSMISSION_ALT_SPEED_TIME_DAY=127
      - TRANSMISSION_ALT_SPEED_TIME_ENABLED=true
      - TRANSMISSION_ALT_SPEED_TIME_END=60
      - TRANSMISSION_ALT_SPEED_UP=0
      - TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false
      - TRANSMISSION_BLOCKLIST_ENABLED=true
      - TRANSMISSION_BLOCKLIST_URL=http://john.bitsurge.net/public/biglist.p2p.gz
I will take a look at your suggested post.

Thank you,

**Comment by rmasri79** [comment]
Hello 
So the good news is I'm now able to download yay not sure what it was but it just started working on its own as of this afternoon.  The bad news is I'm still seeing the connection error in the transmission logs even after adding the pull-filter command into the ovpn file. 
Thank you

**Comment by haugene** [comment]
I'm tempted to close this issue then. It sounds like this is an issue to raise in more Transmission-related forums? Can you also provide your complete logs? In the new image after #846 updates there is no longer a config called Toronto. Are you on the newest image?

**Comment by rmasri79** [comment]
Yes, I'm on the newest image, I missed the config change and just updated it in my compose file.  I'm definitely getting fewer errors in the logs now. I'm attaching a copy of my most recent logs and it looks a lot better than before.  I think you can close this issue sorry to have bothered you with it as you might be correct it looks more like a transmission issue than anything else.  
[logs.txt](https://github.com/haugene/docker-transmission-openvpn/files/3369662/logs.txt)
Thank you very much

**Comment by haugene** [comment]
Hmm. It also looks from your logs that there is a DNS issue:
```
Mon Jul  8 13:49:16 2019 RESOLVE: Cannot resolve host address: ca.torguardvpnaccess.com:1912 (Temporary failure in name resolution),
```

After failing to connect there it falls back to the IP set as the second remote in the .ovpn config for Toronto. Can you exec into a running container and check if you can ping google for example?

I also see that you get some options pushed as well. DNS and ping-stuff. I think you should use ping-filter to ignore those and specify DNS yourself. The ping-flags you get pushed will stop the error handling where Docker restarts the container. I recommend ping-exit instead.

**Comment by rmasri79** [comment]
I wasn't too concerned about the DNS resolve part because I thought that it was a torguard issue. When I connect to the container I'm able to ping google and torguardvpnaccess.com.  

I also started seeing Authentication/Decrypt packet errors as of this morning, so I followed your post #257.

Here is part of my latest yaml file 

Transmission with VPN – Bittorrent Downloader
  transmission-vpn:
    container_name: transmission-vpn
    image: haugene/transmission-openvpn
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun
    restart: always
    ports:
    - "9091:9091"
    dns:
      - 8.8.8.8
      - 8.8.4.4
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ${USERDIR}/docker/transmission-vpn:/data
      - ${USERDIR}/docker/shared:/shared
      - ${USERDIR}/Downloads:/data/watch
      - ${USERDIR}/Downloads/completed:/downloads
      - ${USERDIR}/Downloads/incomplete:/data/incomplete
    environment:
      - OPENVPN_PROVIDER=TORGUARD
      - OPENVPN_USERNAME=
      - OPENVPN_PASSWORD=
      - OPENVPN_CONFIG=Canada.Toronto
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60 --pull-filter ignore ping --mute-replay-warnings
      - LOCAL_NETWORK=192.168.1.0/24
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}

Should I remove the --ping 10 if it will be ignored after?

I'm attaching my latest log file as well.
[logs.txt](https://github.com/haugene/docker-transmission-openvpn/files/3373880/logs.txt)

**Comment by haugene** [comment]
This looks good to me 👍 Nothing wrong in these logs at first glance.
You should keep the --ping 10, as you're ignoring the ping coming from Torguard servers:
```
Tue Jul  9 10:31:13 2019 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 1.1.1.1,dhcp-option DNS 1.0.0.1,route 10.35.0.1,topology net30,ping 5,ping-restart 30,compress,ifconfig 10.35.0.22 10.35.0.21,peer-id 4'
Tue Jul  9 10:31:13 2019 Pushed option removed by filter: 'ping 5'
Tue Jul  9 10:31:13 2019 Pushed option removed by filter: 'ping-restart 30'
```

If you're still getting errors in the Transmission log I think that's out of scope here for now. Everything seems to start and initialise just fine.

**Comment by rmasri79** [comment]
That's great thank you very much for all your help I really appreciate it.

**Comment by haugene** [comment]
You're welcome, and thanks for donating 😄 
Closing this then...
